src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -230,6 +230,9 @@ export class TerminalService implements ITerminalService {
 			env[`PTYSHELLARG${i}`] = arg;
 		});
 		env['PTYCWD'] = this.contextService.getWorkspace() ? this.contextService.getWorkspace().resource.fsPath : os.homedir();
+		if (!env['LANG'] && platform.locale) {
```

Comment:
Some tests for this function would be good too, I can help with this on Monday if you need.